### DISCUSSION TITLE
Is the tutorial code broken?

### DISCUSSION THREAD

**Discussion by christoferd** [solidjs/solid#2218] [Q&A]
Hi, not sure if the tutorial code is broken, or if I'm not understanding it correctly.

https://www.solidjs.com/tutorial/introduction_memos
* Function keeps returning the same number, seems useless

https://www.solidjs.com/tutorial/introduction_effects
* Button click does nothing(?) or seems to do nothing as there is no visual change on the page
 
![image](https://github.com/solidjs/solid/assets/1835883/33faca5d-ef19-4834-b5a2-f373d6de96b3)

![image](https://github.com/solidjs/solid/assets/1835883/858a0662-18ac-4aee-b83e-3b13722ed287)

**Comment by ryansolid** [comment]
No, but you do make a good point in questioning how illustrative those examples are. Just because they are working doesn't make them not confusing.

The `createMemo` example is trying to show reduced cost in calculation. Which is hard to do in a way that is visible on machines of different spec. So each calculation gets that much more expensive, and we did a grid of 40 so that there were enough points to re-use the calculation. So yes they are expected to show the same thing, but after solving it takes higher count before it becomes bogged down.

The `createEffect` example uses console.log instead of rendering to show that work is being done. This is because using `createEffect` to set other state is generally not a good pattern and having it update the DOM immediately so you can see would require introducing other ideas like refs. Instead we use console.log which doesn't display anything in the main screen and only in the console in the dev console.
